processor/src/main/java/org/bindgen/processor/util/ConfUtil.java

```diff
@@ -58,35 +56,58 @@
 	 * So we start there and walk up parent directories looking for
 	 * {@code bindgen.properties} files.
 	 */
-	private static File resolveBindgenPropertiesIfExists(Location location, ProcessingEnvironment env, String fileName) {
-		// Find a dummy /bin/apt/dummy.txt path to start at
-		final String dummyPath;
-		try {
-			// We don't actually create this, we just want its URI
-			FileObject dummyFileObject = env.getFiler().getResource(location, "", "dummy.txt");
-			dummyPath = dummyFileObject.toUri().toString().replaceAll("file:", "");
-		} catch (IOException e1) {
-			return null;
-		}
+	private static InputStream resolveBindgenPropertiesIfExists(Location location, ProcessingEnvironment env, String fileName) {
+	    final String propertiesFilename = "bindgen.properties";
+        final FileObject fileObject;
+        final URL baseUrl;
+        try {
+            fileObject = env.getFiler().getResource(location, "", propertiesFilename);
+            baseUrl = fileObject.toUri()
+                .toURL();
+        } catch (IOException e1) {
+            return null;
+        }
 
-		// Walk up looking for a bindgen.properties
-		File current = new File(dummyPath).getParentFile();
-		while (current != null) {
-			File possible = new File(current, fileName);
-			if (possible.exists()) {
-				return possible;
-			}
-			current = current.getParentFile();
+		String path = baseUrl.getPath();
+        if (path.startsWith("/")) {
+        	path = path.substring(1);
 		}
+		final List<String> componentsWithFile = Arrays.asList(path
+            .split("/"));
+		final List<String> dirPathComponents = componentsWithFile.subList(0, componentsWithFile.size() - 1);
+		final Optional<InputStream> bindingPropertiesInAncestorPaths = IntStream.iterate(dirPathComponents.size(), i -> i - 1)
+            .limit(dirPathComponents.size() - 1)
+            .boxed()
+            .map(i -> dirPathComponents.subList(0, i))
+            .map(list -> list.stream()
+                .collect(Collectors.joining("/", "/", "/" + propertiesFilename)))
+            .<Optional<InputStream>> map(newPath -> {
+                try {
+                    URL url = new URL(baseUrl, newPath);
+                    InputStream inputStream = url.openStream();
```

Comment:
No `try-with-resources` for this `InputStream`? See https://github.com/igloo-project/bindgen/pull/13/files#diff-df832b7eb06ce65f926f349ea1e5c2e3R30 for example.

Reply:
Try-with resources would close the stream at the (possibly implicit) finally part of the try block, but we need to return an `InputStream` that is still open to use at the caller side.

Reply:
May be i'm a too paranoid but what if while computing `bindingPropertiesInAncestorPaths` we open multiple `InputStream`? Then only one will be used and closed properly.

See https://github.com/igloo-project/bindgen/pull/13/files#diff-df832b7eb06ce65f926f349ea1e5c2e3R78

Reply:
There are two cases:

- If the opening of the `InputStream` succeeds it should be returned right away wrapped in an `Optional`, so we open at most one `InputStream`s. The closing of the returned `InputStream` will be the responsibility of the caller, which is done in the [try-with-resources block](https://github.com/igloo-project/bindgen/pull/13/files#diff-df832b7eb06ce65f926f349ea1e5c2e3R31) in the caller method.
- In case the opening of the `InputStream` doesn't succeed, i.e. it throws an exception, we have nothing to close, as the exception itself got thrown while opening the stream, so there's no constructed `InputStream` instance to close. Not even try with resources would help in that case, as it's just a syntactic sugar around `AutoCloseable` instances.

Reply:
Oh ok I got it. For a moment I forgot that thanks to `.findFirst()` we will stop immediately on the first good match. My bad, LGTM.